toolkit/tools/srpmdownloader/srpmdownloader.go

```diff
@@ -0,0 +1,232 @@
+	}
+
+	// Create a template configuration that all packed SRPM will be based on.
+	var templateSrcConfig sourceRetrievalConfiguration
+	// Setup remote source configuration
+	var err error
+	templateSrcConfig.caCerts, err = x509.SystemCertPool()
+	logger.PanicOnError(err, "Received error calling x509.SystemCertPool(). Error: %v", err)
+	if *caCertFile != "" {
+		newCACert, err := ioutil.ReadFile(*caCertFile)
+		if err != nil {
+			logger.Log.Panicf("Invalid CA certificate (%s), error: %s", *caCertFile, err)
+		}
+
+		templateSrcConfig.caCerts.AppendCertsFromPEM(newCACert)
+	}
+
+	if *tlsClientCert != "" && *tlsClientKey != "" {
+		cert, err := tls.LoadX509KeyPair(*tlsClientCert, *tlsClientKey)
+		if err != nil {
+			logger.Log.Panicf("Invalid TLS client key pair (%s) (%s), error: %s", *tlsClientCert, *tlsClientKey, err)
+		}
+
+		templateSrcConfig.tlsCerts = append(templateSrcConfig.tlsCerts, cert)
+
+	}
+	if *srpmUrlList != "" {
+		// Assumes that srpmUrlList come in as ',' seperated
+		urls := strings.Split(*srpmUrlList, " ")
+		templateSrcConfig.sourceURLs = urls
+	}
+
+	// A pack list may be provided, if so only pack this subset.
+	// If none is provided, pack all srpms.
+	srpmList, err := srpm.ParsePackListFile(*srpmListFile)
+	logger.PanicOnError(err)
+
+	logger.Log.Infof("SRPM list %s", srpmList)
+
+	err = getSRPMQueryWrapper(*specsDir, *distTag, *buildDir, *outDir, *workerTar, *workers, *runCheck, srpmList, templateSrcConfig)
+	logger.PanicOnError(err)
+
+}
+
+// getSRPMQueryWrapper wraps getSRPMQuery to conditionally run it inside a chroot.
+// If workerTar is non-empty, packing will occur inside a chroot, otherwise it will run on the host system.
+func getSRPMQueryWrapper(specsDir, distTag, buildDir, outDir, workerTar string, workers int, runCheck bool, srpmList []string, templateSrcConfig sourceRetrievalConfiguration) (err error) {
+	var chroot *safechroot.Chroot
+	originalOutDir := outDir
+	if workerTar != "" {
+		const leaveFilesOnDisk = false
+		chroot, buildDir, outDir, specsDir, err = srpm.CreateSRPMChroot(workerTar, buildDir, outDir, specsDir)
+		if err != nil {
+			return
+		}
+		defer chroot.Close(leaveFilesOnDisk)
+	}
+
+	doCreateAll := func() error {
+		err = getSRPMQuery(specsDir, distTag, buildDir, outDir, workers, runCheck, srpmList, templateSrcConfig)
+		return err
+	}
+
+	if chroot != nil {
+		logger.Log.Info("Grabbing SRPMs URL inside a chroot environment")
+		err = chroot.Run(doCreateAll)
+	} else {
+		logger.Log.Info("Grabbing SRPMs URL SRPMs in the host environment")
+		err = doCreateAll()
+	}
+
+	if err != nil {
+		return err
+	}
+
+	// If this is container build then the bind mounts will not have been created.
+	// Copy the chroot output to host output folder.
+	if !buildpipeline.IsRegularBuild() {
+		srpmsInChroot := filepath.Join(chroot.RootDir(), outDir)
+		err = directory.CopyContents(srpmsInChroot, originalOutDir)
+		if err != nil {
+			return err
+		}
```

Comment:
Simplification idea, but I see benefits of the current approach as well. Just something to consider:
```suggestion
```

Reply:
Alternatively, I'd print an error message saying we failed to perform the copy + print the source and destination paths.